File: backend/workshop/models/employee.py
import uuid
from django.db import models


class Employee(models.Model):
    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
    name = models.CharField(max_length=50)
    email = models.EmailField(unique=True)
    phone = models.CharField(unique=True, max_length=20, blank=True, null=True)
    position = models.CharField(max_length=100, blank=True, null=True)
    address = models.CharField(max_length=255, blank=True, null=True)
    salary = models.DecimalField(max_digits=10, decimal_places=2, blank=True, null=True)
    date_joined = models.DateField(auto_now_add=True)
    is_active = models.BooleanField(default=True)

    class Meta:
        db_table = "employee"

    def __str__(self):
        return f"{self.name}"

File: dashboard/src/types/employee.ts
export interface Employee {
  id: string;
  fullName: string;
  email: string;
  phone: string;
  nic: string;
  position: string;
  address: string;
  joiningDate: string;
  salary: number;
}

export interface CreateEmployee {
  fullName: string;
  email: string;
  phone: string;
  nic: string;
  position: string;
  address: string;
  joiningDate: string;
  salary: number;
}

export interface UpdateEmployee {
  fullName?: string;
  email?: string;
  phone?: string;
  nic?: string;
  position?: string;
  address?: string;
  joiningDate?: string;
  salary?: number;
}
